0322. Coin Change/Solution.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * You are given coins of different denominations and a total amount of money amount.
+ * Write a function to compute the fewest number of coins that you need to make up that amount.
+ * If that amount of money cannot be made up by any combination of the coins, return -1.
+ */
+
+/**
+ * Time Complexity: O(nm)    n is the amount and m is the number of coins
+ * Space Complexity: O(n)
```

Comment:
nice that you reduced the space complexity, I intuitively prefer the matrix filling which gives O(m*n)

Reply:
That would be the intuition but while coding you realize that you don't quite need a matrix.

Reply:
pro @tushar8049